stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/internal/GiphyAttachmentViewHolder.kt

```diff
@@ -0,0 +1,133 @@
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.decorator.internal.BackgroundDecorator
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.decorator.internal.Decorator
+import io.getstream.chat.android.ui.transformer.ChatMessageTextTransformer
+
+internal class GiphyAttachmentViewHolder(
+    parent: ViewGroup,
+    decorators: List<Decorator>,
+    private val listeners: MessageListListenerContainer?,
+    private val markdown: ChatMessageTextTransformer,
+    internal val binding: StreamUiItemGiphyAttachmentBinding = StreamUiItemGiphyAttachmentBinding.inflate(
+        parent.streamThemeInflater,
+        parent,
+        false
+    ),
+) : DecoratedBaseMessageItemViewHolder<MessageListItem.MessageItem>(binding.root, decorators) {
+
+    /**
+     * We override the Message passed to listeners here with the up-to-date Message
+     * object from the [data] property of the base ViewHolder.
+     *
+     * This is required because these listeners will be invoked by the AttachmentViews,
+     * which don't always have an up-to-date Message object in them. This is due to the
+     * optimization that we don't re-create the AttachmentViews when the attachments
+     * of the Message are unchanged. However, other properties (like reactions) might
+     * change, and these listeners should receive a fully up-to-date Message.
+     */
+    private fun modifiedListeners(listeners: MessageListListenerContainer?): MessageListListenerContainer? {
+        return listeners?.let { container ->
+            MessageListListenerContainerImpl(
+                messageClickListener = { container.messageClickListener.onMessageClick(data.message) },
+                messageLongClickListener = { container.messageLongClickListener.onMessageLongClick(data.message) },
+                messageRetryListener = { container.messageRetryListener.onRetryMessage(data.message) },
+                threadClickListener = { container.threadClickListener.onThreadClick(data.message) },
+                attachmentClickListener = { _, attachment ->
+                    container.attachmentClickListener.onAttachmentClick(data.message, attachment)
+                },
+                attachmentDownloadClickListener = container.attachmentDownloadClickListener::onAttachmentDownloadClick,
+                reactionViewClickListener = { container.reactionViewClickListener.onReactionViewClick(data.message) },
+                userClickListener = { container.userClickListener.onUserClick(data.message.user) },
+                giphySendListener = { _, action ->
+                    container.giphySendListener.onGiphySend(data.message, action)
+                },
+                linkClickListener = container.linkClickListener::onLinkClick
+            )
+        }
+    }
+
+    override fun bindData(data: MessageListItem.MessageItem, diff: MessageListItemPayloadDiff?) {
```

Comment:
We have missing comments here in general.

it's okay if this was a conscious decision seeing how a lot of these are self explanatory private methods, I just wanted to note their absence here.